unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            base.Awake();
+            vrmExportButton.onClick.RemoveAllListeners();
+            vrmExportButton.onClick.AddListener(OnVRMExportButtonClick);
+
+            backButton.onClick.RemoveAllListeners();
+            backButton.onClick.AddListener(() => OnBackButtonPressed?.Invoke());
+
+            wearableDetailPool = GetItemDetailPool();
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            var pool = GetItemDetailPool();
+
+            foreach (PoolableObject pooledItem in pooledItems)
+                pool.Release(pooledItem);
+        }
+
+        private Pool GetItemDetailPool()
+        {
+            var entryPool = PoolManager.i.GetPool(VRM_EXPORT_DETAILS_POOL_ID);
+            if (entryPool != null) return entryPool;
+
+            entryPool = PoolManager.i.AddPool(
+                VRM_EXPORT_DETAILS_POOL_ID,
+                Instantiate(vrmDetailPrefab).gameObject,
+                maxPrewarmCount: 5,
+                isPersistent: true);
+
+            entryPool.ForcePrewarm();
+
+            return entryPool;
+        }
+
+        private void OnVRMExportButtonClick()
+        {
+            OnVRMExportButtonPressed?.Invoke();
+        }
+
+        private void EnableVRMExport(bool enable)
+        {
+            vrmExportButton.SetInteractable(enable);
+            exportWarningBar.SetActive(!enable);
+        }
+
+        public override void RefreshControl() { }
+
+        public override void OnDisable()
+        {
+            base.OnDisable();
+
+            foreach (PoolableObject pooledItem in pooledItems)
+                pooledItem.Release();
+        }
+
+        public void FillVRMBlockingWearablesList(List<NFTDataDTO> itemsToDisplay)
+        {
+            EnableVRMExport(false);
+            equippedItemsURNs.Clear();
+
+            foreach (var itemData in itemsToDisplay)
+            {
+                equippedItemsURNs.Add(itemData.urn);
+                var detailItem = wearableDetailPool.Get();
+                detailItem.gameObject.transform.SetParent(wearablesContainer, false);
+                pooledItems.Add(detailItem);
+
+                var detailComponentView = detailItem.gameObject.GetComponent<VRMDetailItemComponentView>();
+
+                if (detailComponentView)
+                {
+                    var vrmItemModel = new VRMItemModel
+                    {
+                        wearableUrn = itemData.urn,
+                        wearableImageUrl = itemData.thumbnail,
+                        wearableName = itemData.name,
+                        wearableCategoryName = itemData.data.wearable.category,
+                        wearableCreatorImageUrl = itemData.creatorImageUrl,
+                        wearableCreatorName = itemData.creatorName,
+                    };
+
+                    detailComponentView.SetModel(vrmItemModel);
+                    detailComponentView.ClearOnWearableUnequippedEvents();
```

Comment:
I think you should also clear the registered events when releasing the item to the pool. It saves undesired memory footprints

Reply:
Agreed, nice suggestion! I also took the opportunity to improve some more this method and the dispose/onDisable, thanks!